KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutCoordinator.swift

```diff
@@ -62,4 +64,75 @@ extension KarhooNewCheckoutCoordinator: NewCheckoutRouter {
         viewController.modalPresentationStyle = .overFullScreen
         baseViewController.present(viewController, animated: true, completion: nil)
     }
+
+    func routeToPassengerDetails(
+        _ currentDetails: PassengerDetails?,
+        delegate: PassengerDetailsDelegate?
+    ) {
+        let detailsViewController = KarhooComponents.shared.passengerDetails(details: currentDetails, delegate: delegate)
+        baseViewController.showAsOverlay(item: detailsViewController, animated: true)
+    }
+
+    func routeSuccessScene(
+        with tripInfo: TripInfo,
+        journeyDetails: JourneyDetails?,
+        quote: Quote,
+        loyaltyInfo: KarhooBookingConfirmationViewModel.LoyaltyInfo
+    ) {
+        if let journeyDetails, journeyDetails.isScheduled {
+            routeToBookingConfirmation(
+                trip: tripInfo,
+                quote: quote,
+                journeyDetails: journeyDetails,
+                loyaltyInfo: loyaltyInfo
+            )
+        } else {
+            callback?(ScreenResult.completed(result: KarhooCheckoutResult(tripInfo: tripInfo)))
+        }
+    }
+
+    /// Dismiss current flow and take user back to booking map
+    func routeToShowBooking() {
+        navigationController?.popToRootViewController(animated: true, completion: {})
+    }
+
+    func routeToBookingConfirmation(
+        trip: TripInfo?,
+        quote: Quote,
+        journeyDetails: JourneyDetails,
+        loyaltyInfo: KarhooBookingConfirmationViewModel.LoyaltyInfo?
+    ) {
+        guard
+            let viewController = viewController,
+            let navigationController = viewController.navigationController
+        else {
+            assertionFailure()
+            return
+        }
+        let masterViewModel = KarhooBottomSheetViewModel(
+            title: UITexts.Booking.prebookConfirmed
+        ) {
+            viewController.dismiss(animated: true, completion: {
+                self.routeToShowBooking()
+            })
+        }
+
+        let slaveViewModel = KarhooBookingConfirmationViewModel(
+            journeyDetails: journeyDetails,
+            quote: quote,
+            trip: trip,
+            loyaltyInfo: loyaltyInfo ?? .init(shouldShowLoyalty: false, loyaltyPoints: 0, loyaltyMode: .none)
+        ) {
+            viewController.dismiss(animated: true, completion: {
+                self.routeToShowBooking()
+            })
+        }
+
+        let screenBuilder = UISDKScreenRouting.default.bottomSheetScreen()
+        let sheet = screenBuilder.buildBottomSheetScreenBuilderForUIKit(viewModel: masterViewModel) {
+            KarhooBookingConfirmationView(viewModel: slaveViewModel)
+        }
+        //        reportBookingConfirmationScreenOpened(tripId: trip?.tripId, quoteId: quote.id)
```

Comment:
Can we uncomment this? Or did you move it somewhere else?

Reply:
I think it should be moved to Booking Confirmation implementation.

Reply:
Fine with me as long as it's not overlooked.